### PR TITLE
Paginate get content collection

### PR DESCRIPTION
**Pull request by steventux**
https://trello.com/c/OZF9rKeX/487-index-endpoint-should-support-pagination

Support pagination and add default ordering of content items index `/v2/content` endpoint.

Original PR here: https://github.com/alphagov/publishing-api/pull/211
It was easier, in light of the large refactor to the Publishing-API, to cherry-pick relevant commits to new branch. Paired with @Rosa-Fox on this story.

### CODE REVIEW

File: app/controllers/v2/content_items_controller.rb [hunk 1]
```diff
@@ -23,6 +24,7 @@ def linkables
           base_path
           internal_name
         ),
+        locale: locale,
```
Comment (tuzz) [inline app/controllers/v2/content_items_controller.rb]: Not sure how this relates?

Comment (steventux) [inline app/controllers/v2/content_items_controller.rb]: I think this is the result of a bad rebase, will fix :+1:

File: spec/controllers/v2/content_items_controller_spec.rb [hunk 2]
```diff
@@ -107,6 +107,46 @@
...
+      end
+      it "is successful" do
+        expect(response.status).to eq(200)
+      end
+      it "responds with the content item as json" do
+        parsed_response_body = JSON.parse(response.body)
+        expect(parsed_response_body.first.fetch("content_id")).to eq("#{content_id}")
+      end
+    end
   end
```
Comment (tuzz) [inline spec/controllers/v2/content_items_controller_spec.rb]: How does the controller pass order params through? (or doesn't it?)

**Comment by tuzz** [comment]
Should we be concerned about the performance impact of ordering by some field? An index would help, but if the user can choose any field we might need indexes on lots of columns?

File: spec/controllers/v2/content_items_controller_spec.rb (hunk 2 above)
Comment (steventux) [inline spec/controllers/v2/content_items_controller_spec.rb]: https://github.com/alphagov/publishing-api/pull/227/files#diff-e23dffac69fdf54083615adffca39488R11 the controller passes a Pagination class initialized with query params to the command class.

**Comment by steventux** [comment]
@tuzz: With regards to ordering, would it be acceptable to fix ordering as the current default : `public_updated_at` for the purposes of introducing pagination and then revisit custom ordering in another story or when the need arises?
I'll add an index to `public_updated_at` if it doesn't already exist.

**Comment by tuzz** [comment]
Seems reasonable. :+1:

**Comment by steventux** [comment]
@tuzz: I've updated with an index on `public_updated_at` and restricted ordering.

**Pull request merged**